x-pack/solutions/security/plugins/security_solution/public/data_view_manager/components/data_view_picker/index.tsx

```diff
@@ -0,0 +1,129 @@
+
+export const DataViewPicker = memo((props: { scope: DataViewManagerScopeName }) => {
+  const dispatch = useDispatch();
+  const selectDataView = useSelectDataView();
+
+  const {
+    services: { dataViewEditor, data, dataViewFieldEditor, fieldFormats },
+  } = useKibana();
+  const closeDataViewEditor = useRef<() => void | undefined>();
+  const closeFieldEditor = useRef<() => void | undefined>();
+
+  const { dataView } = useDataView(props.scope);
+
+  const dataViewId = dataView?.id;
+
+  const createNewDataView = useCallback(() => {
+    closeDataViewEditor.current = dataViewEditor.openEditor({
+      onSave: async (newDataView) => {
+        dispatch(sharedDataViewManagerSlice.actions.addDataView(newDataView));
+        selectDataView({ id: newDataView.id, scope: [props.scope] });
+      },
+      allowAdHocDataView: true,
+    });
+  }, [dataViewEditor, dispatch, props.scope, selectDataView]);
+
+  const onFieldEdited = useCallback(() => {}, []);
+
+  const editField = useCallback(
+    async (fieldName?: string, _uiAction: 'edit' | 'add' = 'edit') => {
+      if (!dataViewId) {
+        return;
+      }
+
+      const dataViewInstance = await data.dataViews.get(dataViewId);
+      closeFieldEditor.current = await dataViewFieldEditor.openEditor({
+        ctx: {
+          dataView: dataViewInstance,
+        },
+        fieldName,
+        onSave: async () => {
+          onFieldEdited();
+        },
+      });
+    },
+    [dataViewId, data.dataViews, dataViewFieldEditor, onFieldEdited]
+  );
+
+  const handleAddField = useCallback(() => editField(undefined, 'add'), [editField]);
+
+  const handleChangeDataView = useCallback(
+    (id: string) => {
+      selectDataView({ id, scope: [props.scope] });
+    },
+    [props.scope, selectDataView]
+  );
+
+  /**
+   * Dispatches an action that will result in data view update in the in-memory caches,
+   * as well as across all scopes where the data view is currently selected.
+   */
+  const handleDataViewModified = useCallback(
+    (updatedDataView: DataView) => {
+      dispatch(sharedDataViewManagerSlice.actions.updateDataView(updatedDataView));
+    },
+    [dispatch]
+  );
+
+  const triggerConfig = useMemo(() => {
+    if (dataView.id === DEFAULT_SECURITY_SOLUTION_DATA_VIEW_ID) {
+      return {
+        label: 'Default Security Data View',
+      };
+    }
+
+    return {
+      label: dataView?.name || dataView?.id || 'Data view',
+    };
+  }, [dataView]);
+
+  const { adhocDataViews: adhocDataViewSpecs, dataViews } = useSelector(sharedStateSelector);
+
+  const savedDataViews = useMemo(() => {
+    const managed: DataViewListItem[] = dataViews.map((spec) => ({
+      id: spec.id ?? '',
+      title: spec.title ?? '',
+      name: spec.name,
+    }));
+
+    return managed;
+  }, [dataViews]);
+
+  const adhocDataViews = useMemo(() => {
+    return adhocDataViewSpecs.map((spec) => new DataView({ spec, fieldFormats }));
```

Comment:
I wonder if we should at the very least store these `adhocDataViewSpecs` in redux (I know it's the opposite of what I said before for the regular dv's... 😓), but since this isn't memoized across dataview picker instances, there may actually be a benefit here... thoughts?

Reply:
oh we do store them, in shared state slice

Reply:
this conversion happens because data view picker component expects the full dv